src/main/java/com/ita/if103java/ims/service/impl/AssociateServiceImpl.java

```diff
@@ -77,18 +89,40 @@ public AssociateDto view(Long id) {
     }
 
     @Override
-    public boolean delete(Long id) {
-        return associateDao.delete(id);
+    public boolean delete(UserDetailsImpl user, Long id) {
+
+        Associate associate = associateDao.findById(id);
+
+        if (associateDao.delete(id)) {
+            EventName eventName = associate.getType() == AssociateType.SUPPLIER ? EventName.SUPPLIER_REMOVED : EventName.CLIENT_REMOVED;
+            createEvent(user, associate, eventName);
+
+            return true;
+        }
+
+        return false;
     }
 
-    private boolean allowToCreateNewAssociate(Long accountId, AssociateType associateType) {
-        Account account = accountDao.findById(accountId);
-        List<Associate> associates = associateDao.findByAccountId(account.getId());
+    private void createEvent(UserDetailsImpl user, Associate associate, EventName eventName) {
+        Event event = new Event();
+        event.setMessage(eventName.getLabel());
```

Comment:
Put here something like this:

[Added,Edited,Removed] [supplier,client] with name <name of the associate>  
for example:
Added supplier with name Supplier ABC